### PR TITLE
APP-454: Redesign price view in offer

### PR DESCRIPTION
**Pull request by hugokallstrom**
[Figma design](https://www.figma.com/file/Ds2hXzDbnUt9ZNhg4HSWH3/Moving-Flow-Android?node-id=216%3A13)

Screenshot: 
device-2021-06-10-144916

### Checklist

- [ ] Functionality is covered by an integration test
- [x] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/util/apollo/GraphQLUtils.kt [hunk 1]
```diff
@@ -39,3 +39,8 @@ fun MonetaryAmount.format(context: Context, market: Market?, minimumDecimals: In
         it.currency = Currency.getInstance(currency.currencyCode)
         it.minimumFractionDigits = minimumDecimals
     }.format(this.number.numberValueExact(BigDecimal::class.java))
+
+fun MonetaryAmount.getCurrencyString(context: Context, market: Market?): String? =
+    NumberFormat.getCurrencyInstance(getLocale(context, market)).also {
+        it.currency = Currency.getInstance(currency.currencyCode)
+    }.currency?.currencyCode
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/util/apollo/GraphQLUtils.kt:46]: Could this function be simplified to `currency.currencyCode`?

File: app/src/main/java/com/hedvig/app/feature/offer/ui/OfferAdapter.kt [hunk 2]
```diff
@@ -92,27 +93,29 @@ class OfferAdapter(
...
-                            val gross =
-                                quote
-                                    .insuranceCost
-                                    .fragments
-                                    .costFragment
-                                    .monthlyGross
-                                    .fragments
-                                    .monetaryAmountFragment
-                                    .toMonetaryAmount()
+                            title.text = title.resources.getString(quote.typeOfContract.getStringId())
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/offer/ui/OfferAdapter.kt:96]:
```suggestion
                            title.setText(quote.typeOfContract.getStringId())
```

File: app/src/main/java/com/hedvig/app/util/apollo/GraphQLUtils.kt (hunk 1 above)
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/util/apollo/GraphQLUtils.kt:46]: Also, I don't think we want to display the ISO-4217-code to users. We're most likely going to want to use `Currency::getSymbol(Locale)`. Sample output from some quick `jshell`-testing:

```java
jshell> import java.util.Currency;

jshell> Currency c = Currency.getInstance("SEK");
c ==> SEK

jshell> c
c ==> SEK

jshell> c.getSymbol();
$4 ==> "kr"

jshell> import java.util.Locale;

jshell> c.getSymbol(Locale.forLanguageTag("en-SE"));
$6 ==> "kr"

jshell> c.getSymbol(Locale.forLanguageTag("en-US"));
$7 ==> "SEK"
```

**Review by hedvigoscar — COMMENTED** [review]
Mostly good, some stuff to be addressed with how we display the currency unit

File: app/src/main/java/com/hedvig/app/util/apollo/GraphQLUtils.kt (hunk 1 above)
Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/util/apollo/GraphQLUtils.kt:46]: Aha! Thanks. Fixed!

Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/util/apollo/GraphQLUtils.kt]: This was also entirely obsoleted by an update in the design. We should now be able to simply use `MonetaryAmount::format` in order to populate the view with the price, and have the per month-label be static 😄

**Review by hedvigoscar — APPROVED** [review]
Approved for merging, unless you want to fix the updated price view in this PR. I trust your judgement!

**Pull request merged**